src/vxingest/prepbufr_to_cb/prepbufr_builder.py

```diff
@@ -0,0 +1,2007 @@
+                    obs_data = self.read_data_from_bufr(
+                        bufr, templates["obs_data_" + str(header_data["report_type"])]
+                    )
+                    subset_data["obs_data"] = obs_data
+
+                    raw_bufr_data[subset_data["station_id"]] = raw_bufr_data.get(
+                        subset_data["station_id"], {}
+                    )
+                    raw_bufr_data[subset_data["station_id"]][
+                        subset_data["report_type"]
+                    ] = raw_bufr_data[subset_data["station_id"]].get(
+                        subset_data["report_type"], {}
+                    )
+                    raw_bufr_data[subset_data["station_id"]][
+                        subset_data["report_type"]
+                    ] = subset_data
+                except Exception as _e:
+                    logger.error(
+                        "PrepBufrBuilder.read_data_from_file: Exception  error: %s",
+                        str(_e),
+                    )
+        bufr.close()
+        return raw_bufr_data
+
+    def build_datafile_doc(self, file_name, data_file_id, origin_type):
+        """
+        This method will build a dataFile document for prepbufr builder. The dataFile
+        document will represent the file that is ingested by the prepbufr builder. The document
+        is intended to be added to the output folder and imported with the other documents.
+        The VxIngest will examine the existing dataFile documents to determine if a specific file
+        has already been ingested.
+        """
+        mtime = Path(file_name).stat().st_mtime
+        df_doc = {
+            "id": data_file_id,
+            "mtime": mtime,
+            "subset": self.subset,
+            "type": "DF",
+            "fileType": "prepbufr",
+            "originType": origin_type,
+            "loadJobId": self.load_spec["load_job_doc"]["id"],
+            "dataSourceId": "NCEP",
+            "url": file_name,
+            "projection": "lambert_conformal_conic",
+            "interpolation": "nearest 4 weighted average",
+        }
+        return df_doc
+
+    def initialize_document_map(self):
+        """
+        reset the document_map for a new file
+        """
+        self.document_map = {}
+
+    def get_document_map(self):
+        """
+        In case there are leftovers we have to process them first.
+        :return: the document_map
+        """
+        try:
+            if len(self.same_time_rows) != 0:
+                self.handle_document()
+            return self.document_map
+        except Exception as _e:
+            logger.exception(
+                "%s get_document_map: Exception in get_document_map: %s",
+                self.__class__.__name__,
+                str(_e),
+            )
+            return None
+
+    # named functions
+    def meterspersecond_to_milesperhour(self, params_dict):
```

Comment:
These kind of handler functions can definitely be made more common. My preferred way to do that is to move them up in the class hierarchy, like put them in common/builder.py since every builder extends that parent class. I prefer that to different modules myself - way less maintenance. As for using Pint we can certainly do that. When the data isn't a Pint.Quantity it has to be converted to one and then converted back to a regular number so it seemed like extra busy work to me. I don't think the other builders rely on Pint very much so it would mean importing it into those classes as well. In the end it doesn't really matter to me. This is pretty straightforward.

Reply:
I think adding the transformation functions to the `common` module would work. Putting them in another module would seem okay to me, if there are enough of them and if it helps with code discovery. Modules are just another folder in the repo. (e.g. `src/vxingest/data_transformations` or something) 

I'm okay either way on Pint, I know we were getting Pint values back from MetPy so could see it making some sense to preserve that information when we extract the data from the NetCDF/GRIB/PREPBUFR files in order to make our lives easier. My hope would be that, if we have the units when we get the raw data, we could specify the output units we want and Pint would handle the unit transformations without us needing to do much, even if the input units change on us. It could be more work than it's worth though.

Reply:
I made an issue for this. I think it should be a separate task

Reply:
Sounds good. And agreed that there's a few steps here - most of which work well in another issue.